examples/features/torch_batch_process/batch_inference/compare_with_core_api/core_api_inference.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import print_function
+
+import json
+import logging
+import math
+import os
+import pathlib
+from typing import Any, Optional
+
+import filelock
+import model
+import torch
+import torch.distributed as dist
+import torchvision as tv
+import torchvision.transforms as transforms
+from torch import nn
+
+import determined as det
+from determined import common, core, pytorch
+
+common.set_logger(False)
+
+
+def run_inference(
+    model: nn.Module,
+    data_loader: Any,
+    context: core.Context,
+    rank: int,
+    skip: int,
+    per_worker_iterate_length: int,
+) -> None:
+    model.eval()
+    records_processed = 0
+
+    inference_output_dir = "inference_out/"
+
+    # The first worker will create it, and exist_ok option makes sure subsequent workers
+    # do not run into error
+    pathlib.Path.mkdir(pathlib.Path(inference_output_dir), parents=True, exist_ok=True)
+
+    dataloader_iterator = iter(data_loader)
+
+    with torch.no_grad():
+        for batch_idx in range(skip, per_worker_iterate_length):
+            X = next(dataloader_iterator, None)
+            logging.info(f"Working on batch is {batch_idx}")
+            if X is not None:
```

Comment:
Reduce indentation?

```
if x is None:
    break
```

Or even better, use `enumerate()` which has a `start` keyword that I only just learned about:

```python
for batch_idx in enumerate(dataloader_iterator, start=skip):
    ...
```

Reply:
See: https://github.com/determined-ai/determined/pull/7274/files#r1263104220